pkg/cmd/server/admission/init.go

```diff
@@ -22,6 +23,7 @@ type PluginInitializer struct {
 	RESTClientConfig      restclient.Config
 	Informers             shared.InformerFactory
 	ClusterQuotaMapper    clusterquotamapping.ClusterQuotaMapper
+	DefaultRegistryFn     imageapi.DefaultRegistryFunc
```

Comment:
> No, this is right, because this is what image streams use to detect
> registry.  The other code is wrong.

That would be the kubelet handling an image pullspec.  If this returns `172.x.y.z` and the image pullspec says `registry.default.svc`, why should only one of those be considered the internal registry?